universal-login-commons/src/core/utils/InitializationHandler.ts

```diff
@@ -0,0 +1,39 @@
+type State<T, B> = {
+  kind: 'Initialized';
+  value: T;
+} | {
+  kind: 'Finalized';
+  value: B | undefined;
+};
+
+export class InitializationHandler<T, B> {
```

Comment:
What happens if you call `initialize` and then immediately call `finalize` while the initialization hasn't completed yet?